src/server/editorServices.ts

```diff
@@ -2083,9 +2102,34 @@ namespace ts.server {
 
             this.printProjects();
 
+            this.telemetryOnOpenFile(fileName, info);
             return { configFileName, configFileErrors };
         }
 
+        private telemetryOnOpenFile(fileName: NormalizedPath, info: ScriptInfo): void {
+            Debug.assert(info.containingProjects.length !== 0);
+            if (
+                !this.eventHandler ||
+                // Don't send for a project with 'checkJs' enabled globally.
+                info.getDefaultProject().getCompilationSettings().checkJs ||

```

Comment:
Shouldnt check this here since you want to add to allJsFilesForOpenFilesTelemetry if the info is js but project doesnt have checkJS